### PR TITLE
add contract whitelist functionality

### PR DESCRIPTION
**Pull request by Timidan**
also added a `genSelectors.js` which i use to generate function selectors at the file level

### CODE REVIEW

File: contracts/RealmDiamond/facets/NFTDisplayFacet.sol [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+pragma solidity 0.8.9;
+
+import "../../libraries/AppStorage.sol";
+
+contract NFTDisplayFacet is Modifiers {
+  event NFTDisplayStatusUpdated(address _token, uint256 _chainId, bool _allowed);
+  //  event NFTBlacklisted(address _token, uint256 _chainId);
+  error LengthMisMatch();
+
+  function whitelistNFTs(address[] calldata _tokens, uint256[] calldata _chainIds) external onlyOwner {
```
Comment (cinnabarhorse) [inline contracts/RealmDiamond/facets/NFTDisplayFacet.sol:11]: We don't need to specify specific tokenIDS here -- the whitelisting is done on a contract-by-contract basis, not specific tokenIDs @Timidan

File: contracts/RealmDiamond/facets/NFTDisplayFacet.sol [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    if (_tokens.length != _chainIds.length) revert LengthMisMatch();
+    for (uint256 i; i < _tokens.length; i++) {
+      s.allowedNFTDisplays[_chainIds[i]][_tokens[i]] = true;
+      emit NFTDisplayStatusUpdated(_tokens[i], _chainIds[i], true);
+    }
+  }
+
+  function blacklistNFTs(address[] calldata _tokens, uint256[] calldata _chainIds) external onlyOwner {
```
Comment (cinnabarhorse) [inline contracts/RealmDiamond/facets/NFTDisplayFacet.sol:19]: This function isn't needed @Timidan

Comment (Timidan) [reply contracts/RealmDiamond/facets/NFTDisplayFacet.sol:19]: Got it

File: contracts/RealmDiamond/facets/NFTDisplayFacet.sol (hunk 1 above)
Comment (cinnabarhorse) [inline contracts/RealmDiamond/facets/NFTDisplayFacet.sol:11]: This function can take a `bool` argument to toggle, instead of having two functions

**Review by cinnabarhorse — CHANGES_REQUESTED** [review]

File: contracts/RealmDiamond/facets/NFTDisplayFacet.sol (hunk 1 above)
Comment (Timidan) [reply contracts/RealmDiamond/facets/NFTDisplayFacet.sol:11]: changing that now

File: scripts/realm/upgrades/upgrade-nftDisplays.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  //TO-DO add remaining addresses
+  const addresses: string[] = [
+    maticAavegotchiDiamondAddress,
+    maticRealmDiamondAddress,
+    maticVars.tileDiamond,
+    maticVars.installationDiamond,
+    aaveFrensArtwork,
+    ghstStakingDiamond,
+  ];
+  //assume all tokens are on matic
```
Comment (cinnabarhorse) [inline scripts/realm/upgrades/upgrade-nftDisplays.ts:46]: Aave Frens Artwork is on ethereum @Timidan

**Review by cinnabarhorse — APPROVED** [review]

**Pull request merged**